src/BrowserDetector.php

```diff
@@ -471,11 +537,18 @@ public static function checkBrowserVivaldi()
     public static function checkBrowserEdge()
     {
         if (stripos(self::$userAgentString, 'Edge') !== false) {
-            $version = explode('Edge/', self::$userAgentString);
-            if (isset($version[1])) {
-                self::$browser->setVersion((float)$version[1]);
+            preg_match('/Edge[\\/ \\(]([a-zA-Z\\d\\.]*)/i', self::$userAgentString, $matches);
+            if (sizeof($matches)>1) {
```

Comment:
We should check if the version is between the range and provide the closest version if it is.

Example:

```
"12.10049" => "0.10.10049",
"12.10051" => "0.11.10051",
```

If we detect version `12.10050`, we should fallback to the latest available version which would be `0.10.10049`. This way, while it may not be 100% accurate, it will provide a fallback to the versions missing in the table.